### DISCUSSION TITLE
Validation of data in Netbox

### DISCUSSION THREAD

**Discussion by dejantep** [netbox-community/netbox#5784] [Ideas]
I'm thinking about implementing some kind of feature to validate netbox data against real network state. Interfaces for instance.
I'm also inexperienced developer but it would be fun project to do.
My idea is to create a report that would get oper-state of all interfaces for given device from my NMS (LibreNMS) and compere this to interface connections in Netbox. This could be a custom button on single device page. When clicked user gets popup displaying results.
Would this be right approach?

**Comment by candlerb** [comment]
There's some previous discussion in #5673.  I think your first step will be to decide how to link a Device or VirtualMachine in Netbox to the corresponding entity in LibreNMS - perhaps using a custom field in Netbox to hold the LibreNMS ID, or by using Netbox's "primary" IP address as the key.

A [custom link](https://netbox.readthedocs.io/en/stable/additional-features/custom-links/) on the device page will certainly work, but it will generate a URL, so will point to a third system (i.e. the system that you write).  Or it could link back to a [custom script](https://netbox.readthedocs.io/en/stable/additional-features/custom-scripts/) on Netbox itself.  Possibly a [report](https://netbox.readthedocs.io/en/stable/additional-features/reports/) could call out to an external system too.

**Reply by dejantep** [reply]
Thanks for the tip. You gave an extensive answer there. Will look into it.
I already have connection between two systems and that is primary ip. Every device in Netbox has custom link that takes user to nms and that particular device.
But as you said URL may need to point to a third system that will do actual validation. Maybe some kind of Flask app.
I never tried napalm implementation in Netbox but if i understood correctly it was kind of what it did.

**Reply by candlerb** [reply]
The napalm interface is quite limited.  In one tab it displays a list of LLDP connections/interfaces ,and colors them according to whether they match Netbox interfaces; in another tab it shows the configuration downloaded from the device.  I don't think Netbox gives an extension point for you to define your own tabs though.
